eigensync/src/lib.rs

```diff
@@ -0,0 +1,284 @@
+                        Some(sender) => sender,
+                        None => {
+                            tracing::error!("No sender for request id {:?}", request_id);
+                            return Ok(());
+                        }
+                    };
+
+                    let _ = sender.send(Err(reason.clone()));
+                },
+            },
+            request_response::Message::Request {
+                request,
+                channel,
+                request_id,
+            } => {
+                tracing::error!("Received request of id {:?}", request_id);
+            }
+        },
+        SwarmEvent::Behaviour(BehaviourEvent::Sync(request_response::Event::OutboundFailure {
+            peer,
+            request_id,
+            error,
+        })) => {
+            let sender = match response_map.remove(&request_id) {
+                Some(sender) => sender,
+                None => {
+                    tracing::error!("No sender for request id {:?}", request_id);
+                    return Ok(());
+                }
+            };
+
+            let _ = sender.send(Err(error.to_string()));
+        }
+        SwarmEvent::ConnectionEstablished { peer_id: _peer_id, .. } => {
+            // send the connection established signal
+            if let Some(sender) = connection_established.take() {
+                let _ = sender.send(());
+            }
+        },
+        other => tracing::error!("Received event: {:?}", other),
+    })
+}
+
+pub struct EigensyncHandle<T: Reconcile + Hydrate + Default + Debug> {
+    pub document: AutoCommit,
+    sender: UnboundedSender<ChannelRequest>,
+    encryption_key: [u8; 32],
+    _marker: PhantomData<T>,
+}
+
+impl<T: Reconcile + Hydrate + Default + Debug> EigensyncHandle<T> {
+    pub async fn new(server_addr: Multiaddr, server_id: PeerId, keypair: identity::Keypair, encryption_key: [u8; 32]) -> anyhow::Result<Self> {
+
+        let mut swarm = SwarmBuilder::with_existing_identity(keypair)
+            .with_tokio()
+            .with_tcp(
+                tcp::Config::default(),
+                noise::Config::new,
+                yamux::Config::default,
+            )
+            .context("Failed to create TCP transport")?
+            .with_behaviour(|_| Ok(client()))
+            .context("Failed to create behaviour")?
+            .with_swarm_config(|cfg| cfg.with_idle_connection_timeout(Duration::MAX))
+            .build();
+                
+        swarm.add_peer_address(server_id.clone(), server_addr.clone());
+
+        swarm.dial(server_id).context("Failed to dial")?;
+        
+        let (sender, receiver) = unbounded_channel();
+        let (connection_ready_tx, connection_ready_rx) = oneshot::channel();
+
+        task::spawn(async move {
+            SyncLoop::new(receiver, swarm, connection_ready_tx).await.run(server_id).await.unwrap();
+        });
+
+        connection_ready_rx.await.context("Failed to establish connection")?;
```

Comment:
We constructor of the eigensync handle shouldn't wait until we get a network connection

If the eigensync server is down, the GUI won't be able to start which is not what we want.

This `initial connection to server -> insert default state if doesn't exist` logic can be moved into `background_sync` and then run in the background.